### ISSUE TITLE
Commandline version dependent on Microsoft.WindowsDesktop.App Framework

### ISSUE THREAD

**Issue by acesnik** [smith-chem-wisc/FlashLFQ#99]
Hi there,

I'm trying to update FlashLFQ in bioconda to use `dotnet`, and it looks like it's dependent on the Microsoft.WindowsDesktop.App Framework that's only available for Windows.

https://github.com/bioconda/bioconda-recipes/pull/28166

https://app.circleci.com/pipelines/github/bioconda/bioconda-recipes/45191/workflows/21a44685-ce3d-43ec-8bf3-fa02914ec5a6/jobs/151515

Digging into FlashLFQ, it looks like this is even the case for `Utils` class library, in addition to `CMD`, and I can't for the life of me figure out where it's dependent on the WindowsDesktop (WPF/Win Forms) namespace. Do you have any clue where to look?

AC

**Comment by acesnik** [comment]
To elaborate on this a bit more, I tried running FlashLFQ from the release on macOS, and I get this message:
```
$ dotnet CMD.dll
It was not possible to find any compatible framework version
The framework 'Microsoft.WindowsDesktop.App', version '3.1.0' was not found.
  - No frameworks were found.

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=3.1.0&arch=x64&rid=osx.10.16-x64
```

**Comment by acesnik** [comment]
That leads here, where the WindowsDesktop framework is only available for Windows:
<img width="533" alt="image" src="https://user-images.githubusercontent.com/16342951/115928011-09de5400-a44b-11eb-8cfd-8f08e4f3b828.png">

**Comment by rmillikin** [comment]
This may be related to an mzLib issue that I mentioned in my review of https://github.com/smith-chem-wisc/mzLib/pull/569 . mzLib may just need to be updated. I'll look into it

**Comment by acesnik** [comment]
That's a good thought. I did try updating mzLib here and wasn't able to make `Util` and `CMD` independent of Windows (targeting net5.0, rather than net5.0-windows).

I briefly tried to figure out if there were any GUI dataframe objects that were used implicitly as input to the methods, but I didn't find any. Do you know of any cases like that?

**Comment by rmillikin** [comment]
I'm fairly sure it's all in .NET Core, there should be no .NET Framework GUI elements. I'm looking into this today, I'll post what I find here

**Comment by rmillikin** [comment]
.NET 5 version of FlashLFQ w/ latest mzLib. Works ok for me on Windows and a Linux docker container. Can check on Ubuntu too if needed. Seems to have no issues w/ targeting. Can you check on Mac?

[FlashLFQ.zip](https://github.com/smith-chem-wisc/FlashLFQ/files/6400882/FlashLFQ.zip)

**Comment by acesnik** [comment]
<img width="773" alt="image" src="https://user-images.githubusercontent.com/16342951/116897803-89f27f80-abea-11eb-959c-ed5f1c616357.png">
Looks good!

**Comment by acesnik** [comment]
Glad to see that merged! I'll wrap up this PR to update FlashLFQ in bioconda after you make a fresh release. 👍🏼  https://github.com/bioconda/bioconda-recipes/pull/28166

**Comment by acesnik** [comment]
Any chance you could make a new patch release so I can wrap up that bioconda PR?

**Comment by bmcnally-uw** [comment]
This still seems to be a problem. The Microsoft.WindowsDesktop.App runtime doesn't seem to be available for Linux, yet CMD.dll appears to require it. Any thoughts?

`$ ./dotnet --list-runtimes
Microsoft.NETCore.App 3.1.0 [/tmp/shared/Microsoft.NETCore.App]
$ ./dotnet CMD.dll
It was not possible to find any compatible framework version
The framework 'Microsoft.WindowsDesktop.App', version '3.1.0' was not found.
  - No frameworks were found.

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=3.1.0&arch=x64&rid=centos.7-x64`

**Comment by wsnoble** [comment]
This problem also arises on MacOS 11.6.

```
% dotnet --version
3.1.416
% dotnet ~/proj/FlashLFQ/CMD.exe 
It was not possible to find any compatible framework version
The framework 'Microsoft.WindowsDesktop.App', version '3.1.0' was not found.
  - No frameworks were found.

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=3.1.0&arch=x64&rid=osx.10.16-x64
```

**Comment by acesnik** [comment]
Rob released version 1.2.1, and the update is now up for review on `bioconda`: https://github.com/bioconda/bioconda-recipes/pull/28166

The updated version is working for me in commandline on macOS 12.2.1, locally:
![image](https://user-images.githubusercontent.com/16342951/157361630-24416730-6602-4fe5-a8ee-322fc57636d9.png)

And it's working on Linux (as well as Mac) in bioconda's build containers:
![image](https://user-images.githubusercontent.com/16342951/157361731-5b62cd84-a154-4bb7-80a0-cfd5b1951fb2.png)

Going to close this for now. Feel free to reopen if you encounter the issue again!

**Comment by trishorts** [comment]
thanks for looking into this anthony. no way that I could have done that
